bluepyemodel/emodel_pipeline/plotting.py

```diff
@@ -1372,49 +1415,103 @@ def plot_FI_curves_comparison(
     # pylint: disable=too-many-nested-blocks, possibly-used-before-assignment
     make_dir(figures_dir)
 
-    emodel_name = None
-    cells = None
-    for emodel in emodels:
-        # do not re-extract data if the emodel is the same as previously
-        if custom_bluepyefe_cells_pklpath is not None:
+    emodel_name, cells = None, None
+
+    def load_cells_for_emodel(emodel):
+        nonlocal cells, emodel_name
+
+        if custom_bluepyefe_cells_pklpath:
             if cells is None:
                 cells = read_extraction_output(custom_bluepyefe_cells_pklpath)
-            if cells is None:
-                continue
-            # experimental FI curve
-            (
-                expt_amp_rel,
-                expt_freq_rel,
-                expt_freq_rel_err,
-                expt_amp,
-                expt_freq_abs,
-                expt_freq_abs_err,
-            ) = get_experimental_FI_curve_for_plotting(cells, prot_name, n_bin=n_bin)
-        elif emodel_name != emodel.emodel_metadata.emodel or cells is None:
-            # take extraction data from pickle file and rearange it for plotting
+            return cells is not None
+
+        if emodel_name != emodel.emodel_metadata.emodel or cells is None:
             cells = read_extraction_output_cells(emodel.emodel_metadata.emodel)
             emodel_name = emodel.emodel_metadata.emodel
-            if cells is None:
-                continue
-
-            # experimental FI curve
-            (
-                expt_amp_rel,
-                expt_freq_rel,
-                expt_freq_rel_err,
-                expt_amp,
-                expt_freq_abs,
-                expt_freq_abs_err,
-            ) = get_experimental_FI_curve_for_plotting(cells, prot_name, n_bin=n_bin)
+            return cells is not None
+        return True
+
+    def get_original_protocol_name(prot_name, evaluator):
+        """Retrieve the protocol name as defined by the user, preserving the original case"""
+        for protocol_name in evaluator.fitness_protocols["main_protocol"].protocols:
+            if prot_name.lower() in protocol_name.lower():
+                return protocol_name
+        return prot_name
+
+    def update_evaluator(expt_amp_rel, prot_name, evaluator):
+        """update evaluator with new simulation protocols."""
+        updated_evaluator = copy.deepcopy(evaluator)
```

Comment:
We don't need this line. In your code, we copy the evaluator before passing it to this function. And this function may be called multiple times. So by removing this line, we prevent BPEM copying over and over the evaluator

Reply:
Right! done in the last commit